tests/test_taskgroups.py

```diff
@@ -179,6 +181,34 @@ async def start_another():
     assert not finished
 
 
+@pytest.mark.parametrize('anyio_backend', ['asyncio'])
```

Comment:
Could you explain the purpose of this test?

Reply:
This is the only test that fails if you don't account for native cancellation in the asyncio `cancelled_caught`. It's the test you requested here: https://github.com/agronholm/anyio/pull/267#discussion_r619647657

Reply:
I just don't see why it has to be this complicated. Isn't starting a single task, natively cancelling it and then checking the status of its task scope enough? Why does the test check if it has started or finished?

Reply:
I had copied the code above. I've now simplified it.